Mention disabling remote cache for Groovy build script compilation in release notes (#28870)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

}
```

> The `buildCache.local.removeUnusedEntriesAfterDays` method is now deprecated.

Reviewer: @asodja, is this related?

> If set to a non-default value, this deprecated setting will take precedence over `Settings.caches.buildCache.setRemoveUnusedEntriesAfterDays()`.

Reviewer: ```suggestion
If this setting is configured with a non-default value, it will take precedence over `Settings.caches.buildCache.setRemoveUnusedEntriesAfterDays()`.
```

}
```

> #### Groovy build script compilation build cache support is disabled

Reviewer: @ljacomet should this be in "Other improvements" or "Build Cache improvements"?
Should these changes be made to the release branch or in the Google doc?

Author: This deprecations were before under `#### Allow version catalog plugin aliases without a version`, but they actually relate to change described under `#### Improved control of local build cache cleanup`. 

So I moved it to that section.

> In Gradle 8.7 we announced that [Groovy build script](userguide/groovy_build_script_primer.html) compilation benefits from the remote build cache.

Reviewer: ```suggestion
In Gradle 8.7, we introduced support for using the remote build cache with [Groovy build script](userguide/groovy_build_script_primer.html) compilation.
```

> Due to reports from users about slower compile times with a remote cache, we did further investigation and found that the cache was not providing the expected performance improvements.

Reviewer: ```suggestion
However, after receiving reports of slower compile times with the remote cache, we conducted further investigation. 
Our findings showed that the cache was not delivering the expected performance improvements.
```

> We disabled the remote build cache for Groovy build script compilation in this release.

Reviewer: ```suggestion
As a result, we have disabled the remote build cache for Groovy build script compilation in this release.
```
